src/measure_theory/integral/integral_eq_improper.lean

```diff
@@ -171,11 +171,26 @@ ae_cover_restrict_of_ae_imp hs
   (λ i, (hφ.measurable i).inter hs)
 
 lemma ae_cover.ae_tendsto_indicator {β : Type*} [has_zero β] [topological_space β]
-  {f : α → β} {φ : ι → set α} (hφ : ae_cover μ l φ) :
+  (f : α → β) {φ : ι → set α} (hφ : ae_cover μ l φ) :
   ∀ᵐ x ∂μ, tendsto (λ i, (φ i).indicator f x) l (𝓝 $ f x) :=
 hφ.ae_eventually_mem.mono (λ x hx, tendsto_const_nhds.congr' $
   hx.mono $ λ n hn, (indicator_of_mem hn _).symm)
 
+lemma ae_cover.ae_measurable {β : Type*} [has_zero β] [metric_space β] [measurable_space β]
+  [borel_space β] [l.is_countably_generated] [l.ne_bot] {f : α → β} {φ : ι → set α}
+  (hφ : ae_cover μ l φ) (hfm : ∀ i, ae_measurable f (μ.restrict $ φ i)) :
+  ae_measurable f μ :=
+begin
+  obtain ⟨u, hu⟩ := l.exists_seq_tendsto,
+  have : ∀ i, ae_measurable ((φ i).indicator f) μ,
+  { intros i,
+    rw ae_measurable_indicator_iff (hφ.measurable i),
+    exact hfm i },
+  refine ae_measurable_of_tendsto_metric_ae (λ n, this $ u n) _,
```

Comment:
You have to use that `β` is a Borel space, since you use `ae_measurable_of_tendsto_metric_ae` here. 
However, I think you can do it without that lemma.
The fact that `ae_measurable ((φ (u n)).indicator f) μ` gives you a sequence of approximations `g n` of `f` on different (but overlapping) subsets. You can iteratively extend the functions `g n` to larger and larger approximations of `f`, something like this:
```
  choose g h1g h2g using this,
  refine ⟨λ x, g (u $ Inf { n | x ∈ φ (u n)}) x, _, _⟩,
```
I think you can prove that this function is measurable and a.e.-equal to `f`. 
I haven't worked out the details, so I could be wrong.

Reply:
Oh, that's much better!